### ISSUE TITLE
Sync Snapshot Failing During Model Transformation

### ISSUE THREAD

**Issue by mjstackhouse** [kontent-ai/data-ops#113]
### Brief bug description

After running the 'sync snapshot' command with 'contentTypes' as an included entity and in an environment whose content types have content type snippet or taxonomy elements, the operation fails while "Transforming the model" with one of the following errors:

`Error: snippet with <snippet id> not found at throwError (<directory location>/node_modules/@kontent-ai/data-ops/build/src/utils/error.js:4:11)\n at transformSnippetElement`

`Error: Could not find taxonomy group (id: <taxonomy group id>) in element(codename: <element codename>)\n at transformTaxonomyElement`

### Repro steps

1. Run the 'sync snapshot' command with 'contentTypes' as an included entity and in an environment whose content types have content type snippet or taxonomy elements
2. See error

### Expected behavior

The 'sync snapshot' command runs successfully with 'contentTypes' as an included entity and in environments whose content types have content type snippet or taxonomy elements.

### Test environment

 - data-ops v2.2.2
 - z shell and Node.js

### Additional context

This error occurs even if the content type snippet or taxonomy group referenced in the content type element currently exists in the environment. I've included screenshots of the types and the corresponding snippet/taxonomy group from the environments where the errors occurred. The command runs successfully as long as the environment's content types don't include content type snippet or taxonomy elements.

### Screenshots

![snippet-error](https://github.com/user-attachments/assets/553ba455-7d5f-4121-897e-931d2531bd7d)
![snippet-error-environment-content-type](https://github.com/user-attachments/assets/6f7bd8dd-eef5-448b-8c02-4a5863f850e4)
![snippet-error-environment-snippet](https://github.com/user-attachments/assets/7cb405fe-b8e5-45f6-8923-b853de50c65f)
![taxonomy-group-error](https://github.com/user-attachments/assets/6fa86b47-e6b1-428c-846a-4142f01f3883)
![taxonomy-error-environment-content-type](https://github.com/user-attachments/assets/74ea832e-74c5-4a3d-bb34-619142a01af1)
![taxonomy-error-environment-taxonomy-group](https://github.com/user-attachments/assets/900fe6fd-74e9-46ea-990d-529ff2e9a747)

**Comment by IvanKiral** [comment]
You are right. The data-ops fetches only the entities from Kontent.ai provided via the `--entities` parameter. This can be fixed by fetching the other needed entities (more mapi calls) or updating the readme to describe those dependencies that also need to be included.

**Comment by mjstackhouse** [comment]
I tested [the fix](https://github.com/kontent-ai/data-ops/pull/115) in an environment with content type snippets and taxonomies being used in the content types (one of the ones that caused the error previously), and yup, it worked!

**Comment by IvanKiral** [comment]
Thanks for the testing. I merged the solution and will be released soon :)
